project-koku/koku
# Resiliency and large OCP payload support for listener

This change better handles large OCP payloads (Addresses #2129) as well as enables retry logic for when the database, kafka restarts, or SIGTERM signal (container down) occurs.  In these cases the last file attempted processing is re-downloaded and processing is attempted again. 

**Testing**
1. Ingest large OCP manifest and stop koku-listener while processing first file.  Verify that on re-start that file is re-downloaded/processed.
2. Ingest large OCP manifest and stop database in the middle of processing a file.  Verify that on db re-start that the file is re-downloaded/processed.
3. Ingest large OCP manifest, verify processing is successful
4. Restart kafka while processing a file. Verify that remaining ingest/process is successful.
5. Create a multi-file payload and verify it is ingested correctly.
6. Create a payload with a file missing but is included in the manifest list.  Verify that 2/3 files are processed with no summary.
7. Corrupt a csv in a payload (add an extra column) and verify that the payload is fails and is committed to the kafka bus

**Test Results**
[ocp_hardening_ut.txt](https://github.com/project-koku/koku/files/4624271/ocp_hardening_ut.txt)
